slave/utils.py

```diff
@@ -240,13 +254,72 @@ def log_banner(text):
     line = "*******************************************************************************\n"
     line += (" " * ((len(line) - 1 - len(text)) / 2)) + text + '\n'
     line += "*******************************************************************************"
-    print line
+    log_info(logging.getLogger(), line)
 
 def make_log(name):
     def log(*args):
-        print "{} -- ".format(name) + ' '.join(args)
+        log_info(logging.getLogger(), "{} -- ".format(name) + ' '.join(args))
     return log
 
 def flush():
     sys.stdout.flush()
     sys.stderr.flush()
+    log_flush()
+
+def create_logger(maxBytes=0, backupCount=0, rollover=False):
+    """Create a rotating file logger which will be used to log messages
+    both to stdout via print as well as to ~/AWFY.log. This will allow
+    us to keep the currently expected print output to stdout while at
+    the same time capturing these printed log messages to a log file
+    for diagnosis.
+
+
+    Use a rotating file handler which will keep the most recent copies
+    of the log file.
+
+    In the main process running task.py, backupCount is set to 3 to
+    keep 3 copies and maxBytes is set to the maximum integer so that
+    the log will not be automatically rolled over. Instead the log
+    will only be rolled over in the main task.py process. Subprocesses
+    which are created will use the same file name but with maxBytes
+    and backupCount set to 0 so they will not roll over at all.
+    """
+    from logging.handlers import RotatingFileHandler
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.DEBUG)
+    file_handler = RotatingFileHandler(
+        os.path.join(os.environ['HOME'], 'AWFY.log'),
```

Comment:
AWFY tasks run on windows too, which don't have `HOME` defined. From Stackoverflow, one way to keep this platform independent is:

```python
import os.path
home_folder = os.path.expanduser('~')
```

Reply:
HOME is available under cygwin. We don't run under the Windows environment at all do we?

Reply:
This comment is for your comment below about CONTROL_TASK_ID. For some reason, I don't have a reply on that comment.

We need CONTROL_TASK_ID to start task.py.  To put CONTROL_TASK_ID into the config file, we'd need to deal with parsing it before calling task.py. I'd rather not at the moment.

Reply:
> HOME is available under cygwin. We don't run under the Windows environment at all do we?

You're right, I missed that; we can keep it as is then.

> To put CONTROL_TASK_ID into the config file, we'd need to deal with parsing it before calling task.py

That's right! Incidentally I didn't change that in the past for the exact same reason. Nevermind my comment then.